inst/pages/98_exercises.qmd

```diff
@@ -1146,13 +1437,119 @@ Useful functions: scater::runMDS, runNMDS, transformAssay, ggplot, scater::plotR
 
 ### Heatmap visualization
 
-1. Load experimental dataset from mia.
-2. Visualize abundances with heatmap.
-3. Visualize abundances with heatmap after CLR + Z transformation.
+1. Load one of the datasets from mia and the miaViz and sechm packages.
+
+```{r}
+#| label: heatmap-ex1
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+library(mia)
+library(miaViz)
+library(sechm)
+
+data("GlobalPatterns", package = "mia")
+tse <- GlobalPatterns
+```
+
+2. Get the most prevalent features. Subset your TreeSe to those features and the first twenty taxa.
 
-See the OMA book for examples.
+```{r}
+#| label: heatmap-ex3
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
 
+most_prevalent <-  names(head(getPrevalence(tse,sort=TRUE)))
 
+tse <- tse[most_prevalent, 1:20]
+```
+
+3. Add a relative abundance assay to the TreeSE.
+
+```{r}
+#| label: heatmap-ex3
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+tse <- transformAssay(tse, assay.type = "counts", method = "relabundance")
+```
+
+5. Visualize the relative abundances with a heatmap using the sechm library
+
+```{r}
+#| label: heatmap-ex5
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+setSechmOption("hmcols", value=c("#F0F0FF","#007562"))
+
+sechm(tse,features = most_prevalent , assayName="relabundance",        show_colnames=TRUE)
+```
+
+### Advanced Heatmaps
+
+1. Import mia, miaviz, complexHeatmap as well as a dataset of your choice
+
+```{r}
+#| label: advanced-heatmap-ex1
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+library(mia)
+library(miaViz)
+library(ComplexHeatmap)
+
+data("Tengeler2020", package = "mia")
+tse <- Tengeler2020
+```
+
+2. Create an assay with relative abundance, then merge the feature by rank Òrder`.
+
+```{r}
+#| label: advanced-heatmap-ex2
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+tse <- transformAssay(tse, assay.type = "counts",
+                             method = "relabundance", pseudocount = 1)
+
+tse <- agglomerateByRank(tse, rank = "Order")
+
+```
+
+3. Add CLR-Z transformation to the TreeSE.
+
+```{r}
+#| label: heatmap-ex6
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+tse<- transformAssay(tse, assay.type = "relabundance",
+                             method = "clr", pseudocount = TRUE)
+
+# Add z-transformation on features (taxa)
+tse <- transformAssay(tse, assay.type = "clr", 
+                             MARGIN = "features",
+                             method = "z", name = "clr_z")
+```
+
+4. plot a heatmap using the CLR-Z transformation
+
+```{r}
+#| label: heatmap-ex6
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+Heatmap(assay(tse, "clr_z"), name = "clr-z")
+```
```

Comment:
The exercise is named as "complex heatmaps". However, this is just basic heatmap. I think we could make this little bit harder.

1. Perform cross-correlation analysis for taxa
2. Visualize with heatmap and add p-values with "X"

Here you can find example on that https://microbiome.github.io/OMA/docs/devel/pages/23_multi-assay_analyses.html#sec-cross-correlation

However, the color of X is bad in the chapter. Can you also change that? Something like this

```
# Create a heatmap and store it
plot <- Heatmap(
    res$cor,
    # Print values to cells
    cell_fun = function(j, i, x, y, width, height, fill) {
        # If the p-value is under threshold
        if( !is.na(res$p_adj[i, j]) & res$p_adj[i, j] < 0.05 ){
            # Print "X"
          grid.shadowtext(sprintf("%s", "X"), x, y, gp = gpar(fontsize = 10, col = "white"))
        }
    },
    heatmap_legend_param = list(title = "", legend_height = unit(5, "cm"))
)
plot
```

Reply:
library("shadowtext")

Reply:
This works, however the p-values are calculated using testExperimentCrossCorrelation(), this gives both the correlation and the p-values but it's deprecated and instead getCrossAssociation() is suggested which only returns the correlation. Any idea on how to get p-values? I can't seem to find anythin in the current documentation.

Reply:
test_significance=TRUE should do the trick